04_library_management_system.rb

```diff
@@ -0,0 +1,125 @@
+class Member
+  attr_accessor :name, :book_issued, :status
+
+  def initialize(name)
+    @name = name
+    @status = "No book taken"
+  end
+end
+
+class Book
+  attr_accessor :name, :issued_to, :status
+
+  def initialize(name)
+    @name = name
+    @status = "New Book"
+  end
+
+  def issue_to_member(member)
+    if (@issued_to.nil?() || @status == "Available")
+      if (member.status == "No book taken")
+        member.status = "Book taken"
+        @issued_to = member
+        @status = "Not Available"
+        puts "#{@name} issued to #{member.name}"
```

Comment:
Avoid puts in methods that are dealing with business logic. Set the state in the method and make a different presentation method to print the status.

Reply:
I added `@message` to my `Book` class and then set it to the respective status_message.
However, since I need to print only one line, instead of defining a new method as follows: 
```
def print_message
  puts @message
end
```
I am returning `@message` from each method and then using `puts` right before the call in my `switch case` to print it.
Is that alright?

Reply:
Seems Good.